lib/spack/spack/test/stage.py

```diff
@@ -766,16 +768,14 @@ def test_get_stage_root_bad_path(self, clear_stage_root):
     )
     def test_stage_purge(self, tmp_path: pathlib.Path, clear_stage_root, path, purged):
         """Test purging of stage directories."""
-        stage_dir = tmp_path / "stage"
-        stage_path = str(stage_dir)
-
-        test_dir = stage_dir / path
-        test_dir.mkdir(parents=True)
-        test_path = str(test_dir)
+        stage_config_path = str(tmp_path / "stage")
 
-        with spack.config.override("config:build_stage", stage_path):
+        with spack.config.override("config:build_stage", stage_config_path):
```

Comment:
This test only ever passed in the past because it relied on `pytest`'s `tmp_path` containing the username of the current user...

In any case, this test is about purging, not about validating the choice of the stage dir, so it should just ask Spack where it put the stage dir after configuring it (and let Spack append a username path component or not)